### PR TITLE
Remove all warnings

### PR DESCRIPTION
**Pull request by crobibero**

### CODE REVIEW

File: Emby.AutoOrganize/Core/NameUtils.cs [hunk 1]
```diff
@@ -80,23 +79,23 @@ private static string GetComparableName(string name)
...
-            .Replace("'", " ")
-            .Replace("[", " ")
-            .Replace("]", " ")
-            .Replace(" a ", string.Empty, StringComparison.OrdinalIgnoreCase)
-            .Replace(" the ", string.Empty, StringComparison.OrdinalIgnoreCase)
-            .Replace(" ", string.Empty);
+            name = name.Replace(".", " ", StringComparison.Ordinal)
+                .Replace("_", " ", StringComparison.Ordinal)
+                .Replace(" and ", " ", StringComparison.Ordinal)
+                .Replace(".and.", " ", StringComparison.Ordinal)
```
Comment (oddstr13) [inline Emby.AutoOrganize/Core/NameUtils.cs]:
```suggestion
                .Replace(" and ", " ", StringComparison.OrdinalIgnoreCase)
                .Replace(".and.", " ", StringComparison.OrdinalIgnoreCase)
```

**Review by oddstr13 — CHANGES_REQUESTED** [review]

**Review by oddstr13 — APPROVED** [review]

File: Emby.AutoOrganize/Core/EpisodeFileOrganizer.cs [hunk 2]
```diff
@@ -880,11 +879,11 @@ private string GetSeriesDirectoryName(Series series, TvFileOrganizationOptions o
...
 
             var seasonFolderName = options.SeriesFolderPattern.
-                Replace("%sn", seriesName)
-                .Replace("%s.n", seriesName.Replace(" ", "."))
-                .Replace("%s_n", seriesName.Replace(" ", "_"))
-                .Replace("%sy", serieYear.ToString())
-                .Replace("%fn", seriesFullName);
+                Replace("%sn", seriesName, StringComparison.Ordinal)
+                .Replace("%s.n", seriesName.Replace(" ", ".", StringComparison.Ordinal), StringComparison.Ordinal)
+                .Replace("%s_n", seriesName.Replace(" ", "_", StringComparison.Ordinal), StringComparison.Ordinal)
```
Comment (Bond-009) [inline Emby.AutoOrganize/Core/EpisodeFileOrganizer.cs]: You can use `string.Replace(char, char)` here

File: Emby.AutoOrganize/Core/EpisodeFileOrganizer.cs [hunk 3]
```diff
@@ -1016,34 +1015,34 @@ private void SetEpisodeFileName(string sourcePath, Series series, Season season,
...
-                .Replace("%0s", seasonNumber.ToString("00", _usCulture))
-                .Replace("%00s", seasonNumber.ToString("000", _usCulture))
-                .Replace("%ext", sourceExtension)
-                .Replace("%en", "%#1")
-                .Replace("%e.n", "%#2")
-                .Replace("%e_n", "%#3")
-                .Replace("%fn", Path.GetFileNameWithoutExtension(sourcePath));
+            var result = pattern.Replace("%sn", seriesName, StringComparison.Ordinal)
+                .Replace("%s.n", seriesName.Replace(" ", ".", StringComparison.Ordinal), StringComparison.Ordinal)
+                .Replace("%s_n", seriesName.Replace(" ", "_", StringComparison.Ordinal), StringComparison.Ordinal)
```
Comment (Bond-009) [inline Emby.AutoOrganize/Core/EpisodeFileOrganizer.cs]: Same

File: Emby.AutoOrganize/Core/MovieFileOrganizer.cs [hunk 4]
```diff
@@ -551,12 +549,12 @@ private string GetMovieNameInternal(string sourcePath, Movie movie, string patte
...
 
-            var result = pattern.Replace("%mn", movieName)
-                .Replace("%m.n", movieName.Replace(" ", "."))
-                .Replace("%m_n", movieName.Replace(" ", "_"))
-                .Replace("%my", productionYear.ToString())
-                .Replace("%ext", sourceExtension)
-                .Replace("%fn", Path.GetFileNameWithoutExtension(sourcePath));
+            var result = pattern.Replace("%mn", movieName, StringComparison.Ordinal)
+                .Replace("%m.n", movieName.Replace(" ", ".", StringComparison.Ordinal), StringComparison.Ordinal)
+                .Replace("%m_n", movieName.Replace(" ", "_", StringComparison.Ordinal), StringComparison.Ordinal)
```
Comment (Bond-009) [inline Emby.AutoOrganize/Core/MovieFileOrganizer.cs]: Same

File: Emby.AutoOrganize/Core/NameUtils.cs [hunk 5]
```diff
@@ -80,23 +79,23 @@ private static string GetComparableName(string name)
...
+                .Replace(")", " ", StringComparison.Ordinal)
+                .Replace(":", " ", StringComparison.Ordinal)
+                .Replace(",", " ", StringComparison.Ordinal)
+                .Replace("-", " ", StringComparison.Ordinal)
+                .Replace("'", " ", StringComparison.Ordinal)
+                .Replace("[", " ", StringComparison.Ordinal)
+                .Replace("]", " ", StringComparison.Ordinal)
+                .Replace(" a ", string.Empty, StringComparison.OrdinalIgnoreCase)
+                .Replace(" the ", string.Empty, StringComparison.OrdinalIgnoreCase)
+                .Replace(" ", string.Empty, StringComparison.Ordinal);
```
Comment (Bond-009) [inline Emby.AutoOrganize/Core/NameUtils.cs]: Same for a lot of these

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Emby.AutoOrganize/Core/NameUtils.cs [hunk 6]
```diff
@@ -78,25 +77,25 @@ private static string GetComparableName(string name)
         {
             name = name.RemoveDiacritics();
 
-            name = " " + name + " ";
+            name = ' ' + name + ' ';
```
Comment (Bond-009) [inline Emby.AutoOrganize/Core/NameUtils.cs]: This isn't actually better since the chars will get boxed for the concat
(haven't benched it yet tho)

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**